### PR TITLE
config: split ListAttribute values on newlines

### PR DESCRIPTION
**Pull request by Exirel**
See the description of #1626 for more information.

Note: I had to upgrade the `test_config.py` file to use py.test instead of unittest. We should be fine with that.

### CODE REVIEW

**Comment by dgw** [comment]
> Note: I had to upgrade the `test_config.py` file to use py.test instead of unittest. We should be fine with that.

Vaguely remembered noticing that `unittest` was used in only that one file… Found where I referenced it over a year ago: https://github.com/sopel-irc/sopel/pull/1299#issue-178875969 (knew I should have made an issue; it did, indeed, get forgotten…)

So, :tada: for nixing that one last mismatched test framework. :grinning:

**Comment by Exirel** [comment]
So, I had to rebase from master because there was several differences in requirements and in documentations needed... and that messed up the commit history in Github.

Talking about documentation, I'll now update it!

**Comment by Exirel** [comment]
@dgw done!

File: sopel/config/types.py [hunk 1]
```diff
@@ -211,12 +211,45 @@ def configure(self, prompt, default, parent, section_name):
...
-    ESCAPE_CHARACTER = '\\'
+        class SpamSection(StaticSection):
+            cheeses = ListAttribute('cheeses')
+
+    the option will be exposed as a Python :class:`list`::
+
+        >>> config.spam.cheeses
+        ['camembert', 'cheddar', 'reblochon']
+
+    which come from this configuration file::
```
Comment (dgw) [inline sopel/config/types.py]:
```suggestion
    which comes from this configuration file::
```

Damn, I think I missed this last time. 🙁

File: sopel/config/types.py [hunk 2]
```diff
@@ -211,12 +211,45 @@ def configure(self, prompt, default, parent, section_name):
...
+
+        [spam]
+        cheeses = camembert
```
Comment (dgw) [inline sopel/config/types.py]: Given that the updated `configuration.rst` examples all place the first list value on the next line, shouldn't the examples in this file do the same?

**Comment by dgw** [comment]
Note to self: Remove #1460 from changelog/milestone when merging this for 7.0, as this PR reverts it.

File: sopel/config/types.py (hunk 2 above)
Comment (Exirel) [reply sopel/config/types.py]: Hm. Probably, yes!

**Pull request merged**